doc/src/manual/noteworthy-differences.md

```diff
@@ -249,7 +249,8 @@ For users coming to Julia from R, these are some noteworthy differences:
   * There are no classes in Julia. Instead there are structures (mutable or immutable), containing data but no methods.
   * Calling a method of a class instance in Python (`x = MyClass(*args); x.f(y)`) corresponds to a function call in Julia, e.g. `x = MyType(args...); f(x, y)`. In general, multiple dispatch is more flexible and powerful than the Python class system.
   * Julia structures may have exactly one abstract supertype, whereas Python classes can inherit from one or more (abstract or concrete) superclasses.
-  * The logical Julia program structure (Packages and Modules) is independent of the file structure (`include` for additional files), whereas the Python code structure is defined by directories (Packages) and files (Modules).
+  * The logical Julia program structure (Packages and Modules) is independent of the file structure, whereas the Python code structure is defined by directories (Packages) and files (Modules).
+  * In Julia, it is idomatic to split the text of large modules into multiple files, without introducing a new module per file. The code is reassembled inside a single module in a main file via `include`. While the Python equivalent (`exec`) is considered harmful (it will silently clobber prior definitions), Julia's compiler prints a warning message if redefinition of types or functions occurs.
```

Comment:
> The code is reassembled inside a single module in a main file via `include`. While the Python equivalent (`exec`) is considered harmful (it will silently clobber prior definitions), Julia's compiler prints a warning message if redefinition of types or functions occurs.


This isn't clear to me. If I define `f() = 1` and I `include` a file that defines `f() = 2`, I can call `f()` without any warning.

Reply:
Maybe what this should stress is that the overwriting behaviour isn't any different in one file vs. several? (If that's true, I think it is.)

I think you get warnings if you overwrite the same signature in a package, but not if you do it at the REPL.

Reply:
> I think you get warnings if you overwrite the same signature in a package, but not if you do it at the REPL.

Right, so maybe this bullet should read:

> The code is reassembled inside a single module in a main file via `include`. The Python equivalent (`execfile`) is considered harmful because it will silently clobber prior function definitions. However, inside a package, Julia's compiler prints a warning message if redefinition of types or functions occurs, which makes `include` safe to use. (At the REPL on the other hand, no error message will be emitted.)

Reply:
`execfile` doesn't exist [since Python 3.0](https://docs.python.org/3.3/whatsnew/3.0.html?highlight=execfile#builtins):

> Removed `execfile()`. Instead of `execfile(fn)` use `exec(open(fn).read())`.

Reply:
Maybe the division between the two bullet points here could be moved a little: The first one says that Julia doesn't care what's in what file/folder, only the code matters, and as a result you should not be surprised to see code in several files. The second one is about overwriting behaviour, which perhaps deserves to be nearby, but isn't quite the same thought.